modern/src/MainPage.js

```diff
@@ -36,39 +41,205 @@ const useStyles = makeStyles((theme) => ({
     },
     overflow: 'hidden',
   },
-  mapContainer: {
-    flexGrow: 1,
+  listContainer: {
+    position: 'absolute',
+    left: theme.spacing(1.5),
+    top: theme.spacing(10.5),
+    width: theme.dimensions.drawerWidthDesktop,
+    zIndex: 1301,
+    height: '100%',
+    maxHeight: `calc(100vh - ${theme.spacing(20)}px)`,
+    [theme.breakpoints.down('md')]: {
+      top: theme.spacing(7),
+      left: '0px',
+      width: '100%',
+      maxHeight: `calc(100vh - ${theme.spacing(14)}px)`,
+    },
+  },
+  paper: {
+    borderRadius: '0px',
+  },
+  toolbar: {
+    paddingLeft: theme.spacing(2),
+    paddingRight: theme.spacing(2),
+  },
+  deviceList: {
+    height: '100%',
+  },
+  collapsed: {
+    transform: `translateX(-${360 + 16}px)`,
+    transition: 'transform .5s ease',
+    [theme.breakpoints.down('md')]: {
+      transform: 'translateX(-100vw)',
+    },
+  },
+  uncollapsed: {
+    transform: `translateX(${theme.spacing(1.5)})`,
+    transition: 'transform .5s ease',
+    [theme.breakpoints.down('md')]: {
+      transform: 'translateX(0)',
+    },
+  },
+  deviceButton: {
+    position: 'absolute',
+    left: theme.spacing(1),
+    top: theme.spacing(10.5),
+    borderRadius: '0px',
+    [theme.breakpoints.down('md')]: {
+      left: theme.spacing(0),
+      top: theme.spacing(7),
+    },
+  },
+  deviceButtonBackground: {
+    backgroundColor: 'white',
+    color: '#777777',
+    '&:hover': {
+      backgroundColor: 'white',
+    },
+  },
+  bottomMenuContainer: {
+    position: 'absolute',
+    left: theme.spacing(1.5),
+    bottom: theme.spacing(1.5),
+    width: theme.dimensions.drawerWidthDesktop,
+    zIndex: 1301,
+    [theme.breakpoints.down('md')]: {
+      bottom: theme.spacing(0),
+      left: '0px',
+      width: '100%',
+    },
+  },
+  menuButton: {
+    display: 'flex',
+    flexDirection: 'column',
+  },
+  iconText: {
+    fontSize: '0.75rem',
+    fontWeight: 'normal',
+    color: '#222222',
```

Comment:
Can you please provide some commentary on the styles. Especially where you have hardcoded numbers and translations.

Reply:
Hardcoded values are used to place the device list container at the absolute position and set the height of the container so that we can show the bottom menu.

Translations -->> In the case of the desktop maximum width of the container is 360 and to hide it is translated x by 360 and in case of tablet and mobile maximum width of the container is 100% and to hide it is translated x by viewport width.

Reply:
Why does it have to be an absolute position?

Reply:
The styles seem to be too complicated. There has to be a way to simplify all of it. @dkyeremeh any ideas?

Reply:
Taking a look

Reply:
To simplify it, we'll need to do a bit of restructuring. I'm working on it. Will create a pr targeting `mail2bishnoi:device_list `

Reply:
@mail2bishnoi I've implemented the changes in https://github.com/mail2bishnoi/traccar-web/pull/1. Kindly merge for the changes to apply here